File: Cpp/942_DI_String_Match.cpp
class Solution {
public:
    vector<int> diStringMatch(string S) {
        vector<int> result;
        int N = S.size();
        int lo = 0;
        int hi = N;
        
        for(int i=0; i<N; ++i){
            if(S[i] == 'I') {
                result.push_back(lo);
                ++lo;
            } else {
                result.push_back(hi);
                --hi;
            }
        }
        result.push_back(lo);
        
        return result;
    }
};

File: Kotlin/942_DI_String_Match.kt
class Solution {
    fun diStringMatch(s: String): IntArray {
        val n = s.length
        val result = IntArray(n + 1)
        var lo = 0
        var hi = n
        
        for (i in s.indices) {
            if(s[i] == 'I') {
                result[i] = lo++
            }else {
                result[i] = hi--
            }
            
        }
        result[n] = lo
        
        return result
    }
}